pkgs/shells/fish/default.nix

```diff
@@ -59,6 +59,9 @@ stdenv.mkDerivation rec {
   '' + ''
     sed -i "s|/sbin /usr/sbin||" \
            "$out/share/fish/functions/__fish_complete_subcommand_root.fish"
+
+    # make fish pick up completions from nix profile
+    printf "set fish_complete_path (printf \$NIX_PROFILES | tr ' ' '\\\\n')\"/share/fish/vendor_completions.d\" \$fish_complete_path" >> $out/share/fish/config.fish
```

Comment:
Because `$NIX_PROFILES` is an external input and `echo` is not consistent, see https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo#65819